### PR TITLE
feat: mapping function

### PR DESCRIPTION
**Pull request by cristianoCatolico**
This pr contains a mapping function called  **MapZapRiskToSeverity** .This function maps a combination of risk level and confidence level (from a web vulnerability scan) to a severity level. This mapping is based on predefined rules that align with CVSS (Common Vulnerability Scoring System) severity ranges.

Inputs:

1. Risk: Represents the risk level of the vulnerability as determined by the scanning tool. Possible values:
- RiskCodeHigh: High risk
- RiskCodeMedium: Medium risk
- RiskCodeLow: Low risk
- RiskCodeInformational: Informational (no real risk)

2. Confidence: Represents the confidence level in the accuracy of the vulnerability detection. Possible values:
- ConfidenceHigh: High confidence in the detection
- ConfidenceFalsePositive: Likely a false positive
- Other confidence levels (e.g., medium, low) are not explicitly handled in this function.

Mapping Logic:

1. High Risk (RiskCodeHigh):
- If confidence is high, the severity is Critical (CVSS 9.0–10.0).
- If confidence is false positive, the severity is downgraded to Medium (CVSS 4.0–6.9).
- Otherwise, the severity is High (CVSS 7.0–8.9).

2. Medium Risk (RiskCodeMedium):

- If confidence is high, the severity is Medium (CVSS 4.0–6.9).
- Otherwise, the severity is Low (CVSS 0.1–3.9).

3. Low Risk (RiskCodeLow):

- If confidence is false positive, the severity is None (CVSS 0.0).
- Otherwise, the severity is Low (CVSS 0.1–3.9).

4. Informational Risk (RiskCodeInformational):

- Always maps to None (CVSS 0.0), as it represents no actionable risk.

5. Default Case:

- If the risk level is unrecognized, the severity is set to Unknown.

### CODE REVIEW

**Comment by grez-lucas** [comment]
`[Good]` Great job with the description @cristianoCatolico ! Logic is clear as water and that made the PR really easy to review 🥇

**Comment by grez-lucas** [comment]
`[Question]` What was your primary source for this logic mapping @cristianoCatolico ?

File: pkg/domain/vulnerability_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			risk:       enums.RiskCodeInformational,
+			confidence: enums.ConfidenceHigh,
+			expected:   enums.SeverityTypeNone,
+		},
+		{
+			name:       "Unknown Risk",
+			risk:       "",
+			confidence: enums.ConfidenceHigh,
+			expected:   enums.SeverityTypeUnknown,
+		},
```
Comment (grez-lucas) [inline pkg/domain/vulnerability_test.go:72]: `[Suggestion]` Although it's the same behavior, we might be missing the test where confidence is an empty string or "Unknown":

 
```suggestion
		{
			name:       "Unknown Risk",
			risk:       "",
			confidence: enums.ConfidenceHigh,
			expected:   enums.SeverityTypeUnknown,
		},
				{
			name:       "Unknown Confidence",
			risk:       enums.RiskCodeInformational,
			confidence: "",
			expected:   enums.SeverityTypeUnknown,
		},
```

**Review by grez-lucas — APPROVED** [review]
LGTM! 👍🏻 Great job with this ticket.

**Comment by cristianoCatolico** [comment]
> `[Question]` What was your primary source for this logic mapping @cristianoCatolico ?

`[Answer]` definitions provided in https://infosec.mozilla.org/guidelines/assessing_security_risk and https://www.zaproxy.org/faq/how-do-i-handle-a-false-positive/ to understand this value. There is no standard, but reading definitions I provided the mappings.
